Loop/Extensions/DeviceDataManager+DeviceStatus.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  DeviceDataManager+DeviceStatus.swift
+//  Loop
+//
+//  Created by Nathaniel Hamming on 2020-07-10.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import LoopKit
+import LoopKitUI
+
+extension DeviceDataManager {
+    var cgmStatusHighlight: DeviceStatusHighlight? {
+        if bluetoothState == .poweredOff {
+            return BluetoothStateManager.bluetoothOffHighlight
+        } else if bluetoothState == .denied ||
+            bluetoothState == .unauthorized
+        {
+            return BluetoothStateManager.bluetoothUnavailableHighlight
+        } else if cgmManager == nil {
+            return DeviceDataManager.addCGMStatusHighlight
+        } else {
+            return (cgmManager as? CGMManagerUI)?.cgmStatusHighlight
+        }
+    }
+    
+    var cgmLifecycleProgress: DeviceLifecycleProgress? {
+        return (cgmManager as? CGMManagerUI)?.cgmLifecycleProgress
+    }
+    
+    var pumpStatusHighlight: DeviceStatusHighlight? {
+        if bluetoothState == .denied ||
```

Comment:
[nit] Why are the Bluetooth conditions and highlight different between CGM and pump? Seems like they should be the same.

Reply:
Yes, I thought they should be as well, but design requires different bluetooth highlights.